gunrock/app/geo/geo_app.cu

```diff
@@ -0,0 +1,376 @@
+// ----------------------------------------------------------------------------
+// Gunrock -- Fast and Efficient GPU Graph Library
+// ----------------------------------------------------------------------------
+// This source code is distributed under the terms of LICENSE.TXT
+// in the root directory of this source distribution.
+// ----------------------------------------------------------------------------
+
+/**
+ * @file hello_app.cu
+ *
+ * @brief Simple Gunrock Application
+ */
+
+#include <gunrock/gunrock.h>
+#include <gunrock/util/test_utils.cuh>
+
+#include <gunrock/graphio/graphio.cuh>
+#include <gunrock/graphio/labels.cuh>
+
+#include <gunrock/app/app_base.cuh>
+#include <gunrock/app/test_base.cuh>
+
+// <DONE> change includes
+#include <gunrock/app/geo/geo_enactor.cuh>
+#include <gunrock/app/geo/geo_test.cuh>
+// </DONE>
+
+namespace gunrock {
+namespace app {
+// <DONE> change namespace
+namespace geo {
+// </DONE>
+
+
+cudaError_t UseParameters(util::Parameters &parameters)
+{
+    cudaError_t retval = cudaSuccess;
+    GUARD_CU(UseParameters_app(parameters));
+    GUARD_CU(UseParameters_problem(parameters));
+    GUARD_CU(UseParameters_enactor(parameters));
+
+    GUARD_CU(parameters.Use<int>(
+        "geo-iter",
+        util::REQUIRED_ARGUMENT | util::SINGLE_VALUE | util::OPTIONAL_PARAMETER,
+        3,
+        "Number of iterations geolocation should run for (default=3).",
+        __FILE__, __LINE__));
+
+    GUARD_CU(parameters.Use<bool>(
+        "geo-complete",
+        util::REQUIRED_ARGUMENT | util::SINGLE_VALUE | util::OPTIONAL_PARAMETER,
+        false,
+        "Run geolocation application until all locations for all nodes are found, uses an atomic (default=false).",
+        __FILE__, __LINE__));
+
+    GUARD_CU(parameters.Use<std::string>(
+        "labels-file",
+        util::REQUIRED_ARGUMENT | util::SINGLE_VALUE | util::OPTIONAL_PARAMETER,
+        "",
+        "User locations label file for geolocation app.",
+        __FILE__, __LINE__));
+
```

Comment:
(low priority) You may want to expose the`max_iter` for `spatial_median` to the user via the command line.  Some of the TA1/2 performers were talking about fixing the number of iterations (to a number ~ 10 instead of ~1000) in order to get an apples-to-apples comparison in the face of floating point errors, etc

Reply:
Done.